### PR TITLE
feat: Set up audit logging basic structure, set up webhook logger, bu…

### PR DESCRIPTION
**Pull request by jonaustin09**
Set up audit logging interface.
Implement audit logging with webhook url.
Bug fix in posix DeleteObject function. Not existing object didn't return error.

### CODE REVIEW

File: s3api/controllers/base.go [hunk 1]
```diff
@@ -765,27 +773,35 @@ func (c S3ApiController) CreateActions(ctx *fiber.Ctx) error {
...
+	Logger s3log.AuditLogger
+	Meta   s3log.LogMeta
 }
 
-func SendResponse(ctx *fiber.Ctx, err error) error {
+func SendResponse(ctx *fiber.Ctx, err error, l *LogOptions) error {
 	if err != nil {
+		if l.Logger != nil {
+			l.Logger.Log(ctx, err, nil, l.Meta)
+		}
```
Comment (benmcclelland) [inline s3api/controllers/base.go]: move this up one line, and then you can remove the success logger call as well

File: s3api/controllers/base.go [hunk 2]
```diff
@@ -800,15 +816,21 @@ func SendResponse(ctx *fiber.Ctx, err error) error {
...
 	return nil
 }
 
-func SendXMLResponse(ctx *fiber.Ctx, resp any, err error) error {
+func SendXMLResponse(ctx *fiber.Ctx, resp any, err error, l *LogOptions) error {
 	if err != nil {
-		fmt.Println(err)
+		if l.Logger != nil {
+			l.Logger.Log(ctx, err, nil, l.Meta)
+		}
```
Comment (benmcclelland) [inline s3api/controllers/base.go:829]: same here, move up one line, remove success case logger

File: s3api/controllers/base_test.go [hunk 3]
```diff
@@ -1359,6 +1362,8 @@ func TestS3ApiController_CreateActions(t *testing.T) {
 	for _, tt := range tests {
 		resp, err := tt.app.Test(tt.args.req)
 
+		fmt.Println(tt.name)
+
```
Comment (benmcclelland) [inline s3api/controllers/base_test.go]: extra output still in here?

File: s3api/controllers/base.go (hunk 2 above)
Comment (mlt180) [reply s3api/controllers/base.go:829]: Here we need the logger separated for error and success cases to get the exact body size

Comment (benmcclelland) [reply s3api/controllers/base.go:829]: oh i see, yes, thats fine

**Review by benmcclelland — APPROVED** [review]

**Pull request merged**